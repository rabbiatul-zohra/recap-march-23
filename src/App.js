import "./styles/App.css";
import React, { useState, useEffect } from "react";
import Grid from "../src/_utils/Grid";
import { BrowserRouter } from "react-router-dom";
import AddCard from "./components/AddCard/AddCard";
import Modal from "../src/_utils/modal/Modal";
import Form from "./components/Form";
import initialData from "../src/data/sampleData.json";
import Artist from "./components/Artist";

const App = () => {
  const [artists, setArtists] = useState(initialData);
  const [artistModelOpen, setArtistModelOpen] = useState(false);

  const handleCloseModal = () => setArtistModelOpen(false);

  const handleAddArtist = (artist) => {
    setArtists([...artists, artist]);
  };

  useEffect(() => {
    console.log("Hello!");
    return () => {
      console.log("Bye!");
    };
  }, []);

  return (
    <BrowserRouter>
      <h1 className='App'>Music Library</h1>
      <Grid>
        {artists.map((artist) => (
          <Artist
            key={artist.id}
            image={artist.image}
            name={artist.artistName}
          />
        ))}
        <AddCard
          text='Add a new artist'
          onClick={() => setArtistModelOpen(true)}
        />
        {artistModelOpen && (
          <Modal onClose={handleCloseModal}>
            <Form
              handleAddArtist={handleAddArtist}
              handleCloseModal={handleCloseModal}
            />
          </Modal>
        )}
      </Grid>
    </BrowserRouter>
  );
};

export default App;
