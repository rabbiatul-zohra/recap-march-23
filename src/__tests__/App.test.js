import { render, screen } from "@testing-library/react";
import App from "../App";
import renderer from "react-test-renderer";

test("renders music library text", () => {
  render(<App />);
  const linkElement = screen.getByText("Music Library");
  expect(linkElement).toBeInTheDocument();
});

test("component renders correctly", () => {
  const rendered = renderer.create(<App />);

  expect(rendered).toBeTruthy();
  expect(rendered).toMatchSnapshot();
});
