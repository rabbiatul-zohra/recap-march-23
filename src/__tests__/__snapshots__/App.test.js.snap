// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders correctly 1`] = `
Array [
  <h1
    className="App"
  >
    Music Library
  </h1>,
  <div
    className="sc-aXZVg jIXfAQ"
  >
    <div
      className="card"
    >
      <img
        alt=""
        className="img"
        src="https://upload.wikimedia.org/wikipedia/commons/1/1c/DuaLipaO2020522_%28101_of_110%29_%2852052470251%29_%28cropped%29.jpg"
      />
      <div
        className="h3"
      >
        Dua Lipa
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt=""
        className="img"
        src="https://upload.wikimedia.org/wikipedia/commons/0/0e/Louis_Armstrong_restored.jpg"
      />
      <div
        className="h3"
      >
        Louis Armstrong
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt=""
        className="img"
        src="https://upload.wikimedia.org/wikipedia/commons/b/b5/191125_Taylor_Swift_at_the_2019_American_Music_Awards_%28cropped%29.png"
      />
      <div
        className="h3"
      >
        Taylor Swift
      </div>
    </div>
    <div
      className="card"
      onClick={[Function]}
      role="button"
    >
      <svg
        className="sc-eqUAAy fOpFXZ"
        height="48"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <title>
          add-circle
        </title>
        <circle
          cx="12"
          cy="12"
          fill="none"
          r="11"
          stroke="#000000"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
        <line
          fill="none"
          stroke="#000000"
          strokeLinecap="round"
          strokeLinejoin="round"
          x1="12"
          x2="12"
          y1="6"
          y2="18"
        />
        <line
          fill="none"
          stroke="#000000"
          strokeLinecap="round"
          strokeLinejoin="round"
          x1="18"
          x2="6"
          y1="12"
          y2="12"
        />
      </svg>
      <h3>
        Add a new artist
      </h3>
    </div>
  </div>,
]
`;
